### PR TITLE
Add missing xPortIsInsideInterrupt function

### PR DESCRIPTION
**Pull request by paulbartell**
Description
-----------
The xPortIsInsideInterrupt function is missing from the ARM Cortex-M0 ports. Add this function.

Test Steps
-----------

Checklist:
----------

- [ X ] I have tested my changes. No regression in existing tests.
- [ N/A ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Skptak — APPROVED** [review]

File: portable/GCC/ARM_CM0/portmacro.h [hunk 1]
```diff
@@ -45,84 +47,121 @@
...
+    {
+        xReturn = pdTRUE;
+    }
+
+    return xReturn;
+}
+
+/*-----------------------------------------------------------*/
+
+#define portMEMORY_BARRIER()    __asm volatile ( "" ::: "memory" )
```
Comment (aggarg) [inline portable/GCC/ARM_CM0/portmacro.h:159]: This is already defined at line #126?

**Review by urutva — APPROVED** [review]

File: portable/IAR/ARM_CM0/portmacro.h [hunk 2]
```diff
@@ -44,87 +47,119 @@
...
+    {
+        xReturn = pdTRUE;
+    }
+
+    return xReturn;
+}
+
+/*-----------------------------------------------------------*/
 
 /* Suppress warnings that are generated by the IAR tools, but cannot be fixed in
```
Comment (n9wxu) [inline portable/IAR/ARM_CM0/portmacro.h:154]: Do we still need to suppress warnings here for IAR when this file is only used by IAR and there are duplicates for the other compilers?

File: portable/GCC/ARM_CM0/portmacro.h (hunk 1 above)
Comment (paulbartell) [inline portable/GCC/ARM_CM0/portmacro.h:159]:
```suggestion
```

File: portable/IAR/ARM_CM0/portmacro.h (hunk 2 above)
Comment (paulbartell) [reply portable/IAR/ARM_CM0/portmacro.h:154]: Yupp. These are still required. The code depends on undefined behavior by not explicitly ordering the accesses to variables declared volatile in some places.

File: portable/GCC/ARM_CM0/portmacro.h (hunk 1 above)
Comment (paulbartell) [reply portable/GCC/ARM_CM0/portmacro.h]: Resolved.

**Review by kstribrnAmzn — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**